### ISSUE TITLE
A "mmengine - WARNING"

### ISSUE THREAD

**Issue by SicongLiu998** [open-mmlab/mmpose#1699]
Thanks for your error report and we appreciate it a lot.
If you feel we have helped you, give us a STAR! :satisfied:

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The bug has not been fixed in the latest version(1.x).

When I try to use the new version to train a model, I met a warning.

10/04 17:52:45 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "data sampler" registry tree. As a workaround, the current "data sampler" registry in "mmengine" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.
10/04 17:52:45 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "optimizer wrapper constructor" registry tree. As a workaround, the current "optimizer wrapper constructor" registry in "mmengine" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.
10/04 17:52:45 - mmengine - WARNING - Failed to search registry with scope "mmpose" in the "optimizer" registry tree. As a workaround, the current "optimizer" registry in "mmengine" is used to build instance. This may cause unexpected failure when running the built modules. Please check whether "mmpose" is a correct scope, or whether the registry is initialized.

- What command or script did you run?
mmpose/configs/animal_2d_keypoint/topdown_heatmap/ap10k/td-hm_hrnet-w32_8xb64-210e_ap10k-256x256.py
```
A placeholder for the command.
```
 bash tools/dist_train.sh /home/lsc/桌面/pose/pose2/mmpose/configs/animal_2d_keypoint/topdown_heatmap/ap10k/td-hm_hrnet-w32_8xb64-210e_ap10k-256x256.py   1

- What config did you run?
mmpose/configs/animal_2d_keypoint/topdown_heatmap/ap10k/td-hm_hrnet-w32_8xb64-210e_ap10k-256x256.py
```
A placeholder for the config.
```

- Did you make any modifications on the code or config? Did you understand what you have modified?
- No, I did not 
- What dataset did you use?
- AP-10K

**Environment**
pytorch 10.0  others following path of the best practice of the 1.x

1. Please run `PYTHONPATH=${PWD}:$PYTHONPATH python mmpose/utils/collect_env.py` to collect necessary environment information and paste it here.
 File "mmpose/utils/collect_env.py", line 2, in <module>
    from mmengine.utils import get_git_hash
  File "/home/lsc/anaconda3/envs/openmmpose/lib/python3.8/site-packages/mmengine/__init__.py", line 3, in <module>
    from .config import *
  File "/home/lsc/anaconda3/envs/openmmpose/lib/python3.8/site-packages/mmengine/config/__init__.py", line 2, in <module>
    from .config import Config, ConfigDict, DictAction
  File "/home/lsc/anaconda3/envs/openmmpose/lib/python3.8/site-packages/mmengine/config/config.py", line 15, in <module>
    from typing import Any, Optional, Sequence, Tuple, Union
  File "/home/lsc/桌面/pose/pose2/mmpose/mmpose/utils/typing.py", line 2, in <module>
    from typing import Dict, List, Optional, Tuple, Union
ImportError: cannot import name 'Dict' from partially initialized module 'typing' (most likely due to a circular import) (/home/lsc/桌面/pose/pose2/mmpose/mmpose/utils/typing.py)

3. You may add addition that may be helpful for locating the problem, such as

- How you installed PyTorch \[e.g., pip, conda, source\]
- Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)
- I have used conda

**Error traceback**

If applicable, paste the error traceback here.

```
A placeholder for traceback.
```

**Bug fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by ly015** [comment]
This warning is probably because that MMPose uses the default data samplers and optimizer wrapper constructors provided by MMEngine for now, and doesn't define these registries under the scope `mmpose`. You can safely ignore these warnings. We will optimize the warning trigger condition.

**Comment by SicongLiu998** [comment]
Thanks a lot for your reply and your great work ! I am looking forward to your stable version of MMPose 1.0.

**Comment by ly015** [comment]
Update: the warning turned out to be triggered by some missing registries in MMPose (the default registry in MMEngine will be used instead with the warning raised). It has been fixed in #1849
